libdrgn/object.c

```diff
@@ -1075,7 +1075,7 @@ drgn_object_cast(struct drgn_object *res,
 		 struct drgn_qualified_type qualified_type,
 		 const struct drgn_object *obj)
 {
-	const struct drgn_language *lang = &drgn_language_c;
+	const struct drgn_language *lang = drgn_type_lang(obj->type);
```

Comment:
For casting, do we want to use the language of the old object or the language of the new type? I think the latter.